✨ OPS-7910: Introduce new logging system using bunyan

s0 opened the pull request:
OPS-7910

This is inspired by what we already have in hpc_service, but has a few changes / enhancements, including:

* Throwing errors when console.*() is used, including a stacktrace
  to identify where it was called from.
* A single shared LogContext used as the root for all logging
* Better module separation + initialization
* Logging JSON output file is specified using env var `LOG_PATH`

s0 committed:
🔧 Fix running of dev server

Following preparing hpc-api for deployment, a change was made to allow
for ts-node compilation to be done for node_modules, but this was not
reflected in the dev script

s0 committed:
➕ Add `bunyan` and `@types/bunyan` as dependencies

s0 committed:
✨ Introduce new logging system using bunyan

This is inspired by what we already have in hpc_service, but has a few
changes / enhancements, including:

* Throwing errors when console.*() is used, including a stacktrace
  to identify where it was called from.
* A single shared LogContext used as the root for all logging
* Better module separation + initialization
* Logging JSON output file is specified using env var `LOG_PATH`

s0 committed:
🔧 Use devServer logging for local dev servers

s0 committed:
Use new logging system for server init message

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

s0 commented:
As noted in OPS-7910, this PR will need to be tested in conjuction with stack changes to make sure logging is working correctly.

s0 committed:
🔧 Set defaults for LOG_MODE and LOG_PATH

s0 committed:
🔧 Check log directory exists before writing

bunyan requires that the parent directory exists before writing the
error log. (this is sensible behavior as we're using docker volume
mounts). To prevent error loops from happening etc...
let's check the existence of the directory before trying to write a log
file there.

s0 committed:
✨ Correctly include stacktraces for unhandled rejection logs

s0 committed:
🔧 Use console.error for when server initialization fails

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
